lib/Runtime/Types/DictionaryTypeHandler.cpp

```diff
@@ -1569,11 +1569,16 @@ namespace Js
                 if (descriptor->GetGetterPropertyIndex() != NoSlots)
                 {
                     *getter = instance->GetSlot(descriptor->GetGetterPropertyIndex());
+                    *setter = nullptr;
                     getset = true;
                 }
                 if (descriptor->GetSetterPropertyIndex() != NoSlots)
                 {
                     *setter = instance->GetSlot(descriptor->GetSetterPropertyIndex());
+                    if(!getset) {
+                        // if we didn't set the getter above, we need to set it here
```

Comment:
`getter` is an out param, right? How can we get away without setting it on all paths here? Why not just assign it and `setter` as well to nullptr right at the top of this function and then we don't care if it was set in above conditional?

Reply:
Due to the _Success_(return) annotation, the outparam annotations are only required to be valid if the function returns true. I agree that it's a bit poorly designed, but the primary goal here was to align things with existing annotations.

Reply:
Alright but I object to the silliness of that annotation.

Reply:
Can we have a getter without a setter, or vice versa? I thought that internally we constructed them in pairs, even if one was a stub "I don't exist, why are you calling me" function.

Reply:
Since `if(!getset)` looks like new code, can we use the existing convention (and preferred by many) of putting a space in `if (`?

Reply:
Also put the `{` on a new line, while you're at it.